pages/api/post/[id].js

```diff
@@ -0,0 +1,13 @@
+import Post from '../../../src/models/Post';
+
+export default async function handler(req, res) {
+	const { id } = req.query;
+
+	if (req.method === 'DELETE') {
+		const deletedPost = await Post.findByIdAndDelete(id);
+		res.status(200).json({ message: 'post deleted', post: deletedPost });
```

Comment:
what happens if the deletion fails? I guess you can add a try and catch block here.